### PR TITLE
Add: codecov to build for test coverage

### LINKED ISSUE #262: Add code cov to test suite run
When we run tests it would be good to see coverage. This won't be full coverage given some of our tests are still only run locally BUT we could get local coverage numbers and then Ci generated numbers too.

### PR DESCRIPTION
**Pull request by lwasser**
closes #262 

This pr adds the code cov integration needed to both push to code cov and get reporting in each PR. 

* you can see the integration in this PR but there's nothing for it to diff against until we merge to master and use it in another PR
* I updated the readme file with a coverage badge as well.

### CODE REVIEW

**Comment by lwasser** [comment]
it just occurred to me that i could also test this by merging into a branch in this organization. let me try that first. i'll convert this to a draft.

**Comment by lwasser** [comment]
ok this is all working. the integration is now visible in this PR. note that code cov can't evaluate a DIFF in coverage yet until we merge this to master. i did update the badge as i can see the repo now in the code cov repo. ready for review!

File: Makefile
Comment (lwasser) [inline Makefile:26]:
```suggestion
	rm -r $(TESTDIR)
```

**Review by jsamoocha — APPROVED** [review]
I don't see any issues with these changes, and there's only one way to find out if it works, so I'll immediately merge this to check if merging #283 will achieve what we want.

**Comment by jsamoocha** [comment]
> ok this is all working. the integration is now visible in this PR. note that code cov can't evaluate a DIFF in coverage yet until we merge this to master. i did update the badge as i can see the repo now in the code cov repo. ready for review!

AFAIU we'll see a diff in coverage once we merge a PR that's based on a parent commit for which we already have coverage. Let me add one trivial test in a PR to quickly check this.

**Pull request merged**
